Build/Run-Tests.ps1

```diff
@@ -3,18 +3,26 @@ $packageConfig = [xml](Get-Content .nuget\packages.config)
 $opencover_version = $packageConfig.SelectSingleNode('/packages/package[@id="OpenCover"]').version
 $opencover_console = "packages\OpenCover.$opencover_version\tools\OpenCover.Console.exe"
 
-&$opencover_console `
-    -register:administrator `
-    -returntargetcode `
-    -hideskipped:All `
-    -filter:"+[*]* -[FluentAssertions*]* -[SmartFormat*]* -[nunit*]*" `
-    -excludebyattribute:*.ExcludeFromCodeCoverage* `
-    -excludebyfile:*\*Designer.cs `
-    -output:"OpenCover.GitExtensions.xml" `
-    -target:"nunit3-console.exe" `
-    -targetargs:"$testAssemblies --workers=1 --timeout=90000"
-$testExitCode = $LastExitCode
-Push-AppveyorArtifact "TestResult.xml"
+$testRunCount = 1
```

Comment:
💡 You can make this a parameter of the script. Example [here](https://github.com/dotnet/roslyn/blob/3864cdfd54619ca30bb52c4f15a221a4af74db68/eng/build.ps1#L17-L63).

Reply:
I agree in principle. But `Run-Test.ps1` is specific to AppVeyor, e.g. it depends on `CodeCov` heavily.
For multiple test runs, the calling script would have to be adapted.

Reply:
We *should* be able to run tests locally, and arguably the only AppVeyor specific script should be in appveyor.yml.

I've been refactoring the pipeline as part of #7396. I'm still not 100% happy even with that state, and I ultimately refactor it using MSBuild props and targets. But it is an involved undertaking and I'm deferring this work until later.

Reply:
Here's how you can run tests locally: https://github.com/gitextensions/gitextensions/pull/7396/commits/b96a8cf5ff81e3b0d8980d14c6b79f6d883ec996

```diff
diff --git a/.nuget/packages.config b/.nuget/packages.config
index 21763ffb7..2d715bad2 100644
--- a/.nuget/packages.config
+++ b/.nuget/packages.config
@@ -2,5 +2,6 @@
 <packages>
   <package id="7-Zip.CommandLine" version="9.20.0" />
   <package id="Codecov" version="1.0.1" />
+  <package id="NUnit.ConsoleRunner" version="3.10.0" />
   <package id="OpenCover" version="4.6.519" />
 </packages>
\ No newline at end of file
```


```diff
diff --git a/Build/Run-Tests.ps1 b/Build/Run-Tests.ps1
index 0ef31407e..64a0414f3 100644
--- a/Build/Run-Tests.ps1
+++ b/Build/Run-Tests.ps1
@@ -1,23 +1,53 @@
$testAssemblies = (Get-ChildItem -Path UnitTests -Filter '*Tests.dll' -Recurse -Exclude 'ApprovalTests.dll').FullName | Where-Object { $_.Contains('\bin\Release') }
 $packageConfig = [xml](Get-Content .nuget\packages.config)
+
+$nunit3Console_version = $packageConfig.SelectSingleNode('/packages/package[@id="NUnit.ConsoleRunner"]').version
+$nunit3Console = "packages\NUnit.ConsoleRunner.$nunit3Console_version\tools\nunit3-console.exe"
+
 $opencover_version = $packageConfig.SelectSingleNode('/packages/package[@id="OpenCover"]').version
 $opencover_console = "packages\OpenCover.$opencover_version\tools\OpenCover.Console.exe"
 
+function Start-Tests {
+    try {
         & $opencover_console `
-    -register:administrator `
-    -returntargetcode `
+            -register:user `
+            -returntargetcode:1000 `
             -hideskipped:All `
             -filter:"+[*]* -[FluentAssertions*]* -[SmartFormat*]* -[nunit*]*" `
             -excludebyattribute:*.ExcludeFromCodeCoverage* `
             -excludebyfile:*\*Designer.cs `
             -output:"OpenCover.GitExtensions.xml" `
-    -target:"nunit3-console.exe" `
-    -targetargs:"$testAssemblies --workers=1 --timeout=90000"
+            -target:"$nunit3Console" `
+            -targetargs:"$testAssemblies --timeout=90000" | Out-Host
+
```

Reply:
Here's how you can run tests locally: https://github.com/gitextensions/gitextensions/pull/7396/commits/708363ef7ed7edde570c1829475a92b2bc8c764e

```diff
diff --git a/.nuget/packages.config b/.nuget/packages.config
index 21763ffb7..2d715bad2 100644
--- a/.nuget/packages.config
+++ b/.nuget/packages.config
@@ -2,5 +2,6 @@
 <packages>
   <package id="7-Zip.CommandLine" version="9.20.0" />
   <package id="Codecov" version="1.0.1" />
+  <package id="NUnit.ConsoleRunner" version="3.10.0" />
   <package id="OpenCover" version="4.6.519" />
 </packages>
\ No newline at end of file
```

```diff
diff --git a/Build/Run-Tests.ps1 b/Build/Run-Tests.ps1
index 0ef31407e..64a0414f3 100644
--- a/Build/Run-Tests.ps1
+++ b/Build/Run-Tests.ps1
@@ -1,23 +1,53 @@
$testAssemblies = (Get-ChildItem -Path UnitTests -Filter '*Tests.dll' -Recurse -Exclude 'ApprovalTests.dll').FullName | Where-Object { $_.Contains('\bin\Release') }
 $packageConfig = [xml](Get-Content .nuget\packages.config)
+
+$nunit3Console_version = $packageConfig.SelectSingleNode('/packages/package[@id="NUnit.ConsoleRunner"]').version
+$nunit3Console = "packages\NUnit.ConsoleRunner.$nunit3Console_version\tools\nunit3-console.exe"
+
 $opencover_version = $packageConfig.SelectSingleNode('/packages/package[@id="OpenCover"]').version
 $opencover_console = "packages\OpenCover.$opencover_version\tools\OpenCover.Console.exe"
 
         & $opencover_console `
-    -register:administrator `
-    -returntargetcode `
+            -register:user `
+            -returntargetcode:1000 `
             -hideskipped:All `
             -filter:"+[*]* -[FluentAssertions*]* -[SmartFormat*]* -[nunit*]*" `
             -excludebyattribute:*.ExcludeFromCodeCoverage* `
             -excludebyfile:*\*Designer.cs `
             -output:"OpenCover.GitExtensions.xml" `
-    -target:"nunit3-console.exe" `
-    -targetargs:"$testAssemblies --workers=1 --timeout=90000"
+            -target:"$nunit3Console" `
+            -targetargs:"$testAssemblies --timeout=90000" | Out-Host
+
```